### PR TITLE
Add classes

### PR DESCRIPTION
**Pull request by FilbertKimasa**
In this pull request, the following was done:

- [ ] Add Person class
- [ ] Add Teacher class
- [ ] Add Student class

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(id, age, parent_permission: true, name: 'Unknown')
+    @id = id
```
Comment (julie-ify) [inline person.rb]: - [ ] **[OPTIONAL]** You can make the id to be an auto-generated variable, so you do not have to pass it in manually. Now, you can remove it from the list of parameters.

  ```
  def initialize(age, parent_permission: true, name: 'Unknown')
    @id = Random.rand(1...1000)
  ```

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Student < Person
+  def initialize(id, classroom, age, parent_permission: true, name: 'Unknown')
+    super(id, age, parent_permission, name)
```
Comment (julie-ify) [inline student.rb]: - [ ] Kindly assign the parameter correctly for the super method.

  ```
  super(id, age, parent_permission: parent_permission, name: name)
  ```

- [ ] **[OPTIONAL]** If you removed the `id` from the Person constructor in the previous comment, then you would need to remove it from lines 4 and 5.

  ```
  def initialize(classroom, age, parent_permission: true, name: 'Unknown')
    super(age, parent_permission: parent_permission, name: name)
  ```

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(id, specialization, age, parent_permission: true, name: 'Unknown')
+    super(id, age, parent_permission, name)
```
Comment (julie-ify) [inline teacher.rb]: - [ ] Kindly assign the parameter correctly for the super method.

  ```
  super(id, age, parent_permission: parent_permission, name: name)
  ```

- [ ] **[OPTIONAL]** If you removed the `id` from the Person constructor in the previous comment, then you would need to remove it from lines 4 and 5.

  ```
  def initialize(specialization, age, parent_permission: true, name: 'Unknown')
    super(age, parent_permission: parent_permission, name: name)
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @FilbertKimasa,

Good job so far!

## Highlights!
- Inheritance of Person class ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Student < Person
+  def initialize(classroom, age, parent_permission: true, name: 'Unknown')
+    super(age: age, parent_permission: parent_permission, name: name)
+    @classroom = classroom
+  end
```
Comment (shahadat3669) [inline student.rb]: Well! great job on updating the parameter for the super method. but still need some updates.
Currently, when I try to create a new Student, I get an error of `/person.rb:5:in `initialize': wrong number of arguments`.  I will recommend you follow the previous reviewer's instruction and follow it fully and update the parameter further.
image
```suggestion
  def initialize(classroom, age, parent_permission: true, name: 'Unknown')
    super(age, parent_permission: parent_permission, name: name)
    @classroom = classroom
  end
```

File: teacher.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(specialization, age, parent_permission: true, name: 'Unknown')
+    super(age: age, parent_permission: parent_permission, name: name)
+    @specialization = specialization
+  end
```
Comment (shahadat3669) [inline teacher.rb]: Well! great job on updating the parameter for the super method. but still need some updates.
Currently, when I try to create a new Teacher, I get an error of `/person.rb:5:in `initialize': wrong number of arguments`.  I will recommend you follow the previous reviewer's instruction and follow it fully and update the parameter further. 🙏 
```suggestion
  def initialize(specialization, age, parent_permission: true, name: 'Unknown')
    super(age, parent_permission: parent_permission, name: name)
    @specialization = specialization
  end
```

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @FilbertKimasa, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ Worked on the previous reviewer's comments.
✅ Updated code



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @FilbertKimasa , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Well done implementing the requested changes. 🎉 🎉 🎉 ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**